From aeb01fb49314017aa70b63531f728e61c983d4b9 Mon Sep 17 00:00:00 2001
From: Matt Turner <mattst88@gmail.com>
Date: Fri, 27 Jan 2023 06:40:05 -0800
Subject: [PATCH] Disable things we don't want

v2: Enable libglad to satisfy egl dependencies
v3: Enable most of libutil to fix undefined references in es2gears
---
 meson.build                   | 11 +++--------
 src/egl/opengl/meson.build    | 33 --------------------------------
 src/egl/opengles2/meson.build |  5 -----
 src/meson.build               |  2 --
 src/util/gl_wrap.h            |  2 --
 src/util/meson.build          |  7 +------
 src/xdemos/meson.build        | 36 -----------------------------------
 7 files changed, 4 insertions(+), 92 deletions(-)

diff --git a/meson.build b/meson.build
index f93a731d..245d1655 100644
--- a/meson.build
+++ b/meson.build
@@ -81,14 +81,7 @@ endif
 
 dep_threads = dependency('threads')
 
-dep_glu = dependency('glu', required : dep_x11.found())
-if not dep_glu.found()
-  _glu_name = 'GLU'
-  if host_machine.system() == 'windows'
-    _glu_name = 'glu32'
-  endif
-  dep_glu = cc.find_library(_glu_name, has_headers: 'GL/glu.h')
-endif
+dep_glu = disabler()
 
 # GBM is needed for EGL on KMS
 dep_gbm = dependency('gbm', required : false, disabler : true)
@@ -127,6 +120,8 @@ else
   dep_glut = dependency('', required : false)
 endif
 
+dep_glut = disabler()
+
 if dep_glut.found() and cc.has_function('glutInitContextProfile',
                                         dependencies: [dep_glut],
                                         prefix : '#include <GL/freeglut.h>')
diff --git a/src/egl/opengl/meson.build b/src/egl/opengl/meson.build
index 434adf21..e4b0c83a 100644
--- a/src/egl/opengl/meson.build
+++ b/src/egl/opengl/meson.build
@@ -25,38 +25,12 @@ executable(
   dependencies: [_deps, dep_glu, idep_eglut_x11],
   install: true
 )
-executable(
-  'egltri_x11', files('egltri.c'),
-  dependencies: [_deps, dep_glu, idep_eglut_x11],
-  install: true
-)
-executable(
-  'xeglgears', files('xeglgears.c'),
-  dependencies: [_deps, dep_glu, dep_egl, dep_x11],
-  install: true
-)
-executable(
-  'xeglthreads', files('xeglthreads.c'),
-  dependencies: [_deps, dep_x11],
-  install: true
-)
 
 executable(
   'eglgears_wayland', files('eglgears.c'),
   dependencies: [_deps, dep_glu, idep_eglut_wayland],
   install: true
 )
-executable(
-  'egltri_wayland', files('egltri.c'),
-  dependencies: [_deps, dep_glu, idep_eglut_wayland],
-  install: true
-)
-
-executable(
-  'eglkms', 'eglkms.c',
-  dependencies: [_deps, dep_drm, dep_gbm, dep_egl],
-  install: true
-)
 
 executable(
   'eglinfo', 'eglinfo.c',
@@ -64,10 +38,3 @@ executable(
   include_directories: [inc_glad, '../../xdemos'],
   install: true
 )
-
-executable(
-  'peglgears', 'peglgears.c',
-  dependencies: [dep_gl, dep_glu, dep_egl, dep_m, idep_util],
-  install: true
-)
-
diff --git a/src/egl/opengles2/meson.build b/src/egl/opengles2/meson.build
index da083cf2..59b35d66 100644
--- a/src/egl/opengles2/meson.build
+++ b/src/egl/opengles2/meson.build
@@ -29,11 +29,6 @@ executable(
   dependencies: [dep_gles2, idep_eglut_x11, idep_util],
   install: true
 )
-executable(
-  'es2tri', files('es2tri.c'),
-  dependencies: [_deps_x11, dep_m],
-  install: true
-)
 executable(
   'es2gears_wayland', files('es2gears.c'),
   dependencies: [dep_gles2, idep_eglut_wayland, idep_util],
diff --git a/src/meson.build b/src/meson.build
index fd4a1673..cea622a6 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -57,5 +57,3 @@ endif
 if host_machine.system() == 'windows'
   subdir('wgl')
 endif
-
-subdir('data')
diff --git a/src/util/gl_wrap.h b/src/util/gl_wrap.h
index b2ff9c8f..f482df5e 100644
--- a/src/util/gl_wrap.h
+++ b/src/util/gl_wrap.h
@@ -7,10 +7,8 @@
 
 #ifdef __APPLE__
 #  include <OpenGL/gl.h>
-#  include <OpenGL/glu.h>
 #else
 #  include <GL/gl.h>
-#  include <GL/glu.h>
 #endif
 
 #ifndef GLAPIENTRY
diff --git a/src/util/meson.build b/src/util/meson.build
index ddcd4834..72e9480d 100644
--- a/src/util/meson.build
+++ b/src/util/meson.build
@@ -21,17 +21,12 @@
 inc_util = include_directories('.')
 
 files_libutil = files(
-  'readtex.c',
   'showbuffer.c',
   'trackball.c',
   'matrix.c',
 )
 
-_deps = [dep_glu, dep_m]
-if dep_glut.found()
-  files_libutil += files('shaderutil.c')
-  _deps += dep_glut
-endif
+_deps = [dep_m]
 
 _libutil = static_library(
   'util',
diff --git a/src/xdemos/meson.build b/src/xdemos/meson.build
index d6d5d5d5..ce26699f 100644
--- a/src/xdemos/meson.build
+++ b/src/xdemos/meson.build
@@ -21,25 +21,7 @@
 glx_deps = [dep_gl, dep_x11, dep_m]
 
 progs = [
-  'glsync',
-  'glxdemo',
   'glxgears',
-  'glxgears_pixmap',
-  'glxcontexts',
-  'glxheads',
-  'glxpixmap',
-  'glxpbdemo',
-  'glxsnoop',
-  'glxswapcontrol',
-  'manywin',
-  'multictx',
-  'offset',
-  'overlay',
-  'shape',
-  'sharedtex',
-  'texture_from_pixmap',
-  'wincopy',
-  'xfont',
 ]
 foreach p : progs
   executable(
@@ -56,23 +38,7 @@ executable(
   install: true
 )
 
-executable(
-  'xrotfontdemo',
-  files('xrotfontdemo.c', 'xuserotfont.c'),
-  dependencies: glx_deps,
-  install: true
-)
-
-_libpbutil = static_library(
-  'pbutil',
-  files('pbutil.c'),
-  dependencies: glx_deps
-)
-
 pbutil_progs = [
-  'glxgears_fbconfig',
-  'pbinfo',
-  'pbdemo',
 ]
 foreach p : pbutil_progs
   executable(
@@ -84,8 +50,6 @@ foreach p : pbutil_progs
 endforeach
 
 thread_progs = [
-  'glthreads',
-  'sharedtex_mt',
 ]
 foreach p : thread_progs
   executable(
